lib/ansible/modules/windows/win_data_deduplication.ps1

```diff
@@ -0,0 +1,174 @@
+$settings = Get-AnsibleParam -obj $params -name 'settings' -type 'list' -default $null
+$dedup_job = Get-AnsibleParam -obj $params -name 'dedup_job' -type 'list' -default $null
+
+# In theory it works on Windows 2012, but this was only developed to work
+# from 2012R2 and higher.
+if ([System.Environment]::OSVersion.Version -lt [Version]"6.3.9600.0") {
+    Fail-Json -message "win_data_deduplication requires Windows Server 2012R2 or higher."
+}
+
+$result = @{
+  changed = $false
+  reboot_required= $false
+}
+
+function Set-DataDeduplication($volume, $state, $settings, $dedup_job) {
+
+  $current_state = 'disabled'
+
+  try {
+    $dedup_info = Get-DedupVolume -Volume "$($volume.DriveLetter):"
+  } catch {
+    $dedup_info = $null
+  }
+
+  if($null -ne $dedup_info) {
+    if ($dedup_info.Enabled) {
+      $current_state = 'enabled'
+    }
+  } else {
+    $current_state = $null
+  }
+
+  if ( $state -ne $current_state ) {
+    if($state -eq 'enabled') {
+      try {
+        # Enable-DedupVolume -Volume <String>
+        Enable-DedupVolume -Volume "$($volume.DriveLetter):"
+      } catch {
+        Fail-Json $result $_.Exception.Message
+      }
+    } elseif ($state -eq 'disabled') {
+      try {
+        Disable-DedupVolume -Volume "$($volume.DriveLetter):"
+      } catch {
+        Fail-Json $result $_.Exception.Message
+      }
+    } else {
+      Fail-Json $result "Unsupported state option, it can only be 'enabled' or 'disabled'."
+    }
+
+    $result.msg += " Deduplication on volume: "+$volume.DriveLetter+" set to: $state"
+    $result.changed = $true
+  }
+
+  if ($state -eq 'enabled') {
+    if ($null -ne $settings) {
+      Set-DataDedupJobSettings -volume $volume -settings $settings
+    }
+
+    if ($null -ne $dedup_job) {
+      Start-DataDedupJob -volume $volume -dedup_job $dedup_job
+    }
+  }
+}
+
+function Set-DataDedupJobSettings ($volume, $settings) {
+
+  try {
+    $dedup_info = Get-DedupVolume -Volume "$($volume.DriveLetter):"
+  } catch {
+    $dedup_info = $null
+  }
+
+  ForEach ($key in $settings.keys) {
+    $update_key = $key
+    $update_value = $settings.$($key)
+
+    if ($update_key -eq "MinimumFileSize" -and $update_value -lt 32768) {
+      $update_value = 32768
+    }
+
+    try {
+      $current_value = ($dedup_info | Select-Object -ExpandProperty $update_key)
+    } catch {
+      $current_value = $null
+    }
+
+    if ($update_value -ne $current_value) {
+      $command_param = @{
+        $($update_key) = $update_value
+      }
+
+      # Set-DedupVolume -Volume <String>`
+      #                 -NoCompress <bool> `
+      #                 -MinimumFileAgeDays <UInt32> `
+      #                 -MinimumFileSize <UInt32> (minimum 32768)
+      try {
+        Set-DedupVolume -Volume "$($volume.DriveLetter):" @command_param
+        $result.msg += " Setting DedupVolume settings for $update_key"
+
+      } catch {
+        Fail-Json $result $_.Exception.Message
```

Comment:
No need for the catch block, just remove it altogether. The same applies for pretty much all your other try/catch blocks that just call `Fail-Json`. Only use try/catch if you either need to catch a specific exception and work with that or the existing exception is quite vague and you can improve upon it.